### PR TITLE
Support Elasticsearch 8 clusters

### PR DESCRIPTION
**Pull request by cmark**
This PR adds the ability to connect to Elasticsearch 8 clusters via the API compatibility mode using the now deprecated high-level Java REST client.
The PR also adds the ability to test against Elasticsearch 8 with https://www.testcontainers.org/modules/elasticsearch/

### CODE REVIEW

File: commons/com.b2international.index.test.tools/.classpath [hunk 1]
```diff
@@ -1,5 +1,14 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <classpath>
+	<classpathentry exported="true" kind="lib" path="lib/annotations-17.0.0.jar"/>
+	<classpathentry exported="true" kind="lib" path="lib/commons-compress-1.21.jar"/>
+	<classpathentry exported="true" kind="lib" path="lib/docker-java-api-3.2.13.jar"/>
+	<classpathentry exported="true" kind="lib" path="lib/docker-java-transport-3.2.13.jar"/>
+	<classpathentry exported="true" kind="lib" path="lib/docker-java-transport-zerodep-3.2.13.jar"/>
+	<classpathentry exported="true" kind="lib" path="lib/duct-tape-1.0.8.jar"/>
+	<classpathentry exported="true" kind="lib" path="lib/elasticsearch-1.17.1.jar" sourcepath="C:/Users/mark/.m2/repository/org/testcontainers/elasticsearch/1.17.1/elasticsearch-1.17.1-sources.jar"/>
```
Comment (apeteri) [inline commons/com.b2international.index.test.tools/.classpath]: The source attachment includes a local directory (see also two lines below).

File: commons/com.b2international.index.test.tools/src/com/b2international/index/IndexResource.java [hunk 2]
```diff
@@ -59,13 +72,39 @@ private IndexResource(Collection<Class<?>> types, Consumer<ObjectMapper> objectM
...
 			mapper.setVisibility(PropertyAccessor.FIELD, Visibility.ANY);
-			client = Indexes.createIndexClient(UUID.randomUUID().toString(), mapper, new Mappings(), indexSettings.get());
+			client = Indexes.createIndexClient(UUID.randomUUID().toString(), mapper, new Mappings(), settings);
 			index = new DefaultIndex(client);
 			revisionIndex = new DefaultRevisionIndex(index, new TimestampProvider.Default(), mapper);
 		}
 		
+		if (container != null) {
+			// make sure we update the synonyms.txt inside the test container
+			container.copyFileToContainer(MountableFile.forHostPath(EsIndexClientFactory.DEFAULT_PATH.resolve(IndexClientFactory.DEFAULT_CLUSTER_NAME).resolve(EsNode.CONFIG_DIR).resolve(EsNode.SYNONYMS_FILE)), "/usr/share/elasticsearch/config/" + EsNode.SYNONYMS_FILE);
```
Comment (apeteri) [inline commons/com.b2international.index.test.tools/src/com/b2international/index/IndexResource.java]: Please split this line into more parts.

File: commons/com.b2international.index.tests/build.properties [hunk 3]
```diff
@@ -1,4 +1,12 @@
...
-               .
+               .,\
+               lib/annotations-17.0.0.jar,\
+               lib/docker-java-api-3.2.13.jar,\
+               lib/docker-java-transport-3.2.13.jar,\
+               lib/docker-java-transport-zerodep-3.2.13.jar,\
+               lib/duct-tape-1.0.8.jar,\
+               lib/elasticsearch-1.17.1.jar,\
+               lib/jna-5.8.0.jar,\
+               lib/testcontainers-1.17.1.jar
```
Comment (apeteri) [inline commons/com.b2international.index.tests/build.properties]: Is this addition also needed in `index.tests`? It looks like `index.test.tools` is the project which copies dependencies to its `lib` folder.

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/EvaluateQueryRefSetRequest.java [hunk 4]
```diff
@@ -47,20 +47,9 @@ public final class EvaluateQueryRefSetRequest extends IndexResourceRequest<Branc
...
-				})
-				.transform(new Function<Request<BranchContext, QueryRefSetMemberEvaluation>, QueryRefSetMemberEvaluation>() {
-					@Override
-					public QueryRefSetMemberEvaluation apply(Request<BranchContext, QueryRefSetMemberEvaluation> input) {
-						return input.execute(context);
-					}
-				})
+				getQueryMembers(context, referenceSet).stream()
+				.map(input -> SnomedRequests.prepareQueryRefSetMemberEvaluation(input.getId()).setLocales(locales()).setExpand(expand()).build())
+				.map(input -> input.execute(context))
```
Comment (apeteri) [inline snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/EvaluateQueryRefSetRequest.java]: These two mapping steps could be merged together; is there a reason they are kept separate?

File: commons/com.b2international.index/src/com/b2international/index/mapping/DocumentMapping.java [hunk 5]
```diff
@@ -157,21 +156,17 @@ public DocumentMapping(Class<?> type, DocumentMapping parent, boolean nested) {
...
 				}
 			})
 			.filter(fieldType -> isNestedDoc(fieldType))
-			.toMap(new Function<Class<?>, DocumentMapping>() {
-				@Override
-				public DocumentMapping apply(Class<?> input) {
-					return new DocumentMapping(input, DocumentMapping.this.parent == null ? DocumentMapping.this : DocumentMapping.this.parent, true);
-				}
-			});
+			.distinct()
```
Comment (apeteri) [inline commons/com.b2international.index/src/com/b2international/index/mapping/DocumentMapping.java:168]: Good catch! It looks like `FluentIterable#toMap` uses a `LinkedHashMap` internally while building the final (immutable) result, and so silently "de-duplicated" the keys. 👍

**Review by apeteri — CHANGES_REQUESTED** [review]

File: commons/com.b2international.index.tests/build.properties (hunk 3 above)
Comment (apeteri) [reply commons/com.b2international.index.tests/build.properties]: This comment appears to have arrived late 🥷

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/EvaluateQueryRefSetRequest.java (hunk 4 above)
Comment (cmark) [reply snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/EvaluateQueryRefSetRequest.java]: No, probably a leftover from an earlier state.

**Review by apeteri — APPROVED** [review]
LGTM!

**Pull request merged**